backend/accounts/baseaccount.go

```diff
@@ -65,9 +67,10 @@ type BaseAccount struct {
 }
 
 // NewBaseAccount creates a new Account instance.
-func NewBaseAccount(config *AccountConfig, log *logrus.Entry) *BaseAccount {
+func NewBaseAccount(config *AccountConfig, coin coin.Coin, log *logrus.Entry) *BaseAccount {
```

Comment:
So, can I provide a config with say `eth-erc20-usdt` code and a btc coin? If so, maybe it would make sense to enforce only compatible combinations? This is where a NewXxx function actualle comes in handy.

Reply:
A code is only an identifier, anything is valid. The caller of this function decides what to identify the accounts by.

The opposite would also be possible, to enumerate all possibilities here, and move them out from the backend. Not sure if that would be helpful, sounds less modular.

Reply:
Ah ok. I thought account code had some more strict meaning. Maybe rename it to just that in another PR, `Identifier`? to avoid any doubts.